### PR TITLE
Add CustomRecovery builtin middleware 

### PR DESCRIPTION
**Pull request by theonlyjohnny**
- With pull requests:
  - Open your pull request against `master`
  - Your pull request should have no more than two commits, if not you should squash them.
  - It should pass all tests in the available continuous integration systems such as TravisCI.
  - You should add/modify tests to cover your proposed code changes.
  - If your pull request contains a new feature, please document it on the README.

CustomRecovery allows the user to control what happens in the case of a panic without having to re-implement the Recovery middleware

### CODE REVIEW

**Comment by theonlyjohnny** [comment]
:wave: Could I get a maintainer's :eyes: on this plz? Currently using my fork and would like to not have to

File: recovery.go [hunk 1]
```diff
@@ -26,13 +26,26 @@ var (
...
 	return RecoveryWithWriter(DefaultErrorWriter)
 }
 
+//CustomRecovery returns a middleware that recovers from any panics and calls the provided handle func to handle it.
+func CustomRecovery(handle RecoveryFunc) HandlerFunc {
+	return CustomRecoveryWithWriter(DefaultErrorWriter, handle)
+}
+
 // RecoveryWithWriter returns a middleware for a given writer that recovers from any panics and writes a 500 if there was one.
 func RecoveryWithWriter(out io.Writer) HandlerFunc {
```
Comment (appleboy) [inline recovery.go:43]: maybe `func RecoveryWithWriter(out io.Writer, ...RecoveryFunc)`?

Comment (theonlyjohnny) [reply recovery.go:43]: wanted to leave a function so the end user didn't have to think about which writer to use, but adding optional RecoveryFunc to RecoveryWithWriter is a good idea -- adding that

**Comment by elftech** [comment]
why still no CustomRecovery in the recovery.go when I download v1.6.3 using go module?

**Pull request merged**